test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,331 @@
+		err := waitForPodRunningInNamespace(c, pod.Name, pod.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func verifyPodsListining(c *client.Client, ns, name string, pods *api.PodList) {
+	By("trying to dial each unique pod")
+	retryTimeout := 2 * time.Minute
+	retryInterval := 5 * time.Second
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	err := wait.Poll(retryInterval, retryTimeout, podResponseChecker{c, ns, label, name, pods}.checkAllResponses)
+	if err != nil {
+		Failf("Did not get expected responses within the timeout period of %.2f seconds: %v", retryTimeout.Seconds(), err)
+	}
+}
+
+var _ = Describe("ResizeNodes", func() {
+	supportedProviders := []string{"gce"}
+	var testName string
+	var c *client.Client
+	var ns string
+	var testingNs *api.Namespace
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err = createTestingNS("resize-nodes", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !providerIs(supportedProviders...) {
+			return
+		}
+		By(fmt.Sprintf("destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+		By(fmt.Sprintf("restoring the original node instance group size"))
+		if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original cluster size: %s", err)
+		}
+	})
+
+	testName = "should be able to delete nodes."
+	It(testName, func() {
+		Logf("starting test %v", testContext.CloudConfig.NumNodes)
+
+		if !providerIs(supportedProviders...) {
+			By(fmt.Sprintf("Skipping %q test, which is only supported for providers %v (not %s)",
+				testName, supportedProviders, testContext.Provider))
+			return
+		}
+
+		if testContext.CloudConfig.NumNodes < 2 {
+			By(fmt.Sprintf("skipping %q test, which requires at lease 2 nodes (not %v)",
+				testName, testContext.CloudConfig.NumNodes))
+			return
+		}
+
+		// Create a replication controller for a service that serves its hostname.
+		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+		name := "my-hostname-delete-node-" + string(util.NewUUID())
+		replicas := testContext.CloudConfig.NumNodes
+		createServeHostnameReplicationController(c, ns, name, replicas)
+		pods := waitForPodsCreated(c, ns, name, replicas)
+		waitForPodsRunning(c, pods)
+		verifyPodsListining(c, ns, name, pods)
+
+		By(fmt.Sprintf("decreasing cluster size to %v", replicas-1))
+		err := resizeNodeInstanceGroup(replicas - 1)
```

Comment:
Please change all of these to the form:

```go
if err := doSomething(...); err != nil {
  Failf("Oh no: %v", err)
}
```

This is idiomatic Go and restricts variables to the tightest scope.